presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java

```diff
@@ -36,6 +39,28 @@ public DockerContainer(String dockerImageName)
         setCopyToFileContainerPathMap(new LinkedHashMap<>());
     }
 
+    @Override
+    public DockerContainer withFileSystemBind(String hostPath, String containerPath)
+    {
+        verifyHostPath(hostPath);
+        return super.withFileSystemBind(hostPath, containerPath);
+    }
+
+    @Override
+    public DockerContainer withFileSystemBind(String hostPath, String containerPath, BindMode mode)
+    {
+        verifyHostPath(hostPath);
+        return super.withFileSystemBind(hostPath, containerPath, mode);
+    }
+
+    // Mounting a non-existing file results in docker creating a directory. This is often not the desired effect. Fail fast instead.
+    private static void verifyHostPath(String hostPath)
```

Comment:
Can you use this in copy methods also? I recently was copying non existing file and it failed very late (during product tests execution).

Reply:
added (where possible)